NWNXLib/Trace.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include "Platform/Time.hpp"
+#include <cstdio>
+
+namespace NWNXLib {
+
+namespace Trace {
+
+#define TRACE_DEBUG(format, ...) \
```

Comment:
Personal preference, but I'd favor simpler `DEBUG()`, `WARNING()`, `ERROR()`, etc macros. The TRACE part isn't really contributing anything useful.

Reply:
I don't like DEBUG() WARNING() etc because it's not clear what those things actually do. To me, TRACE_ makes it clearer - we could change it to LOG_ if that's better? Shorter and more accurate than TRACE_ I guess. What do you think?

Reply:
I guess LOG_ works better. Or PRINT_. To me, TRACE would sooner be another logging level, like/below DEBUG. But I don't feel very strongly about this.